crypt_test.go

```diff
@@ -1212,6 +1212,100 @@ func (s *cryptSuite) TestActivateVolumeWithRecoveryKeyErrorHandling8(c *C) {
 	})
 }
 
+type testActivateVolumeWithKeyData struct {
+	activateOptions []string
+	keyData         []byte
+	expectedKeyData []byte
+	errMatch        string
+	cmdCalled       bool
+}
+
+func (s *cryptSuite) testActivateVolumeWithKey(c *C, data *testActivateVolumeWithKeyData) {
+	c.Assert(data.keyData, NotNil)
+
+	expectedKeyDataPath := filepath.Join(c.MkDir(), "expected-key")
+	mockSdCryptsetup := snapd_testutil.MockCommand(c, c.MkDir()+"/systemd-cryptsetup", fmt.Sprintf(`
+set -x
+key=$(xxd -p < "$4")
+if [ ! -f "%[1]s" ]; then echo "test error: expected key data not mocked"; exit 123; fi
+expected="$(xxd -p < "%[1]s")"
+if [ "$key" != "$expected" ]; then
+    exit 5
+fi
```

Comment:
This suite already has a mock for systemd-cryptsetup, with 2 expected keys set up by the suite - tpmKey and recoveryKey, although I have a refactoring in another branch that renames tpmKey but it should be ok to use here despite the name. Is it possible to use the existing mock rather than creating a new one?